Usbipd.PowerShell/GetUsbipdDevice.cs

```diff
@@ -0,0 +1,80 @@
+﻿// SPDX-FileCopyrightText: 2022 Frans van Dorsselaer
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+using System.Diagnostics;
+using System.IO;
+using System.Management.Automation;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Usbipd.Automation;
+
+namespace Usbipd.PowerShell
+{
+    [Cmdlet(VerbsCommon.Get, "UsbipdDevice")]
+    [OutputType(typeof(Device))]
+    public class GetUsbipdDeviceCommand : Cmdlet
+    {
+        State State = new();
+
+        protected override void BeginProcessing()
+        {
+            WriteDebug($"Detected installation path: {Installation.ExePath}");
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = Installation.ExePath,
+                UseShellExecute = false,
+                StandardOutputEncoding = Encoding.UTF8,
+                StandardErrorEncoding = Encoding.UTF8,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                Arguments = "state",
+            };
+
+            using var process = Process.Start(startInfo);
+            if (process is null)
+            {
+                throw new ApplicationFailedException($"Cannot execute '{Installation.ExePath}'.");
+            }
+
+            var stdout = string.Empty;
+            var stderr = string.Empty;
+
+            var captureTasks = new[]
+            {
+                Task.Run(async () => { stdout = await process.StandardOutput.ReadToEndAsync(); }),
+                Task.Run(async () => { stderr = await process.StandardError.ReadToEndAsync(); }),
+            };
+
+            process.WaitForExit();
```

Comment:
I think this is safe as-is, but if you want you can use `WaitForExitAsync()` and add the resulting `Task` to `captureTasks`.

Reply:
Unfortunately, `WaitForExitAsync()` isn't part of of `netstandard2.0`, which we need for this to work on PowerShell versions.

Reply:
Got it, I missed that dependency.